### PR TITLE
"make check" should always clean up... wait and try again if rm fails

### PR DESCRIPTION
**Pull request by awallace-cray**

### CODE REVIEW

File: util/test/checkChplInstall [hunk 1]
```diff
@@ -104,6 +104,15 @@ log_info "Temporary test job directory: ${TMP_TEST_DIR}"
 function cleanup_tmp_dir()
 {
     log_debug "Removing ${TMP_TEST_DIR}"
+    local i
+    # sometimes on an NFS file system, rm needs to wait for the TEST_JOB exe file to close
+    for i in 1 2 3 4 5
```
Comment (ben-albrecht) [inline util/test/checkChplInstall]: is 5 tries just a 'magic number' that happens to work well here?

File: util/test/checkChplInstall [hunk 2]
```diff
@@ -104,6 +104,15 @@ log_info "Temporary test job directory: ${TMP_TEST_DIR}"
...
+    do
+        rm -rf ${TMP_TEST_DIR} && break
+        log_warning "failed rm -rf ${TMP_TEST_DIR}, wait and try again"
+        ( set -x; ls -la ${TMP_TEST_DIR} ) || : ok
+        sleep 1
+    done
     rm -rf ${TMP_TEST_DIR} || (
```
Comment (ben-albrecht) [inline util/test/checkChplInstall]: Do we still need to remove the `${TMP_TEST_DIR} after the retry loop from above?

Comment (awallace-cray) [inline util/test/checkChplInstall]: If rm -rf fails 5 times, the loop ends. 
In that case, that final rm -rf will produce an error message that says we give up now. Thats all its good for.

File: util/test/checkChplInstall (hunk 1 above)
Comment (awallace-cray) [inline util/test/checkChplInstall]: 5 is an arbitrary guess that seems reasonable.  
We will see how many tries it really takes when some Jenkins jobs run on the Crays after this PR merges. Thats the only way to know.

Comment (ben-albrecht) [inline util/test/checkChplInstall]: Got it. I'd prefer a `seq 5` here if this is something we're going to be tuning, but we can find out if this is something that we'll be tuning or not after a night of testing.

**Pull request merged**